feature-libs/quote/components/comments/quote-comments.component.ts

```diff
@@ -0,0 +1,209 @@
+      .pipe(
+        take(1),
+        // do for error and success
+        finalize(() => this.commentsComponent.resetForm())
+      )
+      .subscribe(
+        // success
+        () => {
+          this.eventService.dispatch({}, QuoteDetailsReloadQueryEvent);
+          this.messagingConfigs.newMessagePlaceHolder = undefined;
+        },
+        // error
+        () => {
+          this.translationService
+            .translate('quote.comments.invalidComment')
+            .pipe(take(1))
+            .subscribe(
+              (text) => (this.messagingConfigs.newMessagePlaceHolder = text)
+            );
+        }
+      );
+  }
+
+  /**
+   * Ensures that the quote item list is expanded, before scrolling to the corresponding item.
+   *
+   * @param event - click event containing the target item
+   */
+  onItemClicked(event: { item: MessageEventBoundItem }) {
+    this.quoteItemsComponentService.setQuoteEntriesExpanded(true);
+    this.quoteItemsComponentService
+      .getQuoteEntriesExpanded()
+      .pipe(take(1), delay(0)) // delay this task until the UI has expanded
+      .subscribe(() => {
+        const aTags = Array.from(this.document.getElementsByTagName('a'));
+        for (const aTag of aTags) {
+          if (aTag.textContent === event.item.name) {
+            aTag.scrollIntoView({ block: 'center' });
+            return;
+          }
+        }
+      });
+  }
+
+  /**
+   * Checks whether the comment section shall be shown at all.
+   * In case the quote is in readonly mode and no comments are present, comment section is hidden, as there is nothing to display.
+   * In case the quote is editable the comment section is always shown, as you could at least add comments from it.
+   *
+   * @param quote - quote
+   * @returns true, only is the comment section shall be shown
+   */
+  showComments(quote: Quote): boolean {
+    let numItemComments = 0;
+    quote.entries?.forEach(
+      (entry) => (numItemComments += entry.comments?.length ?? 0)
+    );
+    return (
+      quote.isEditable ||
+      (quote.comments?.length ?? 0) > 0 ||
+      numItemComments > 0
+    );
+  }
+
+  protected prepareMessagingConfigs(): MessagingConfigs {
+    return {
+      charactersLimit:
+        this.quoteUIConfig.quote?.maxCharsForComments ??
+        DEFAULT_COMMENT_MAX_CHARS,
+      displayAddMessageSection: this.quoteDetails$.pipe(
+        map((quote) => quote.isEditable)
+      ),
+      dateFormat: 'MMMM d, yyyy h:mm aa',
+      defaultItemId: ALL_PRODUCTS_ID,
+      itemList$: this.prepareItemList(),
+      sendBtnIsNotPrimary: true,
+    };
+  }
+
+  protected prepareItemList(): Observable<MessageEventBoundItem[]> {
+    let allProducts: string = 'quote.comments.allProducts';
+    this.translationService
+      .translate(allProducts)
+      .pipe(take(1))
+      .subscribe((text) => (allProducts = text));
+
+    return this.quoteDetails$.pipe(
+      map((quote) => {
```

Comment:
[MEDIUM]
The `allProducts` variable might be undefined, if the `translationService.translat()` emits a value with a delay - it is the case when a customer configures to use lazy loadable translation resources.

Instead, whenever getting translations, please first prefer `| cxTranslate` pipe in HTML. Or if it's really justified to get translationsin TS, then use `combineLatest` rxjs operator. For instance:

```ts
return combineLatest([
  this.quoteDetails$,
  this.translationService.translate('quote.comments.allProducts')]).pipe(
    map(([quote, allProductsLabel]) => {
      // use both `quote` and `allProductsLabel`
    }
  )
```

Reply:
@Uli-Tiger

Reply:
thx - refactored accordingly